docs/sphinx/source/reference/sql_types.rst

```diff
@@ -72,14 +72,172 @@ In this example, `c` is an array, and each record within the array is a struct o
+* :sql:`VECTOR(N, HALF)`: 1 + (2 × N) bytes
+* :sql:`VECTOR(N, FLOAT)`: 1 + (4 × N) bytes
+* :sql:`VECTOR(N, DOUBLE)`: 1 + (8 × N) bytes
+
+Working with Vectors
+====================
+
+Vector Literals and Prepared Statements
+----------------------------------------
+
+**Important**: Vector literals are not directly supported in SQL. Vectors must be inserted using **prepared statement parameters** through the JDBC API.
+
+In the JDBC API, you would create a prepared statement and bind vector parameters using the appropriate Java objects (e.g., :java:`HalfRealVector`, :java:`FloatRealVector`, or :java:`DoubleRealVector`):
+
+.. code-block:: java
+
+    // Java JDBC example
+    PreparedStatement stmt = connection.prepareStatement(
+        "INSERT INTO embeddings VALUES (?, ?)");
+    stmt.setLong(1, 1);
+    stmt.setObject(2, new FloatRealVector(new float[]{0.5f, 1.2f, -0.8f}));
+    stmt.executeUpdate();
+
+For documentation and testing purposes, the examples below use a special test DSL syntax :sql:`!! !vXX [values] !!` to represent prepared statement parameters:
+
+* :sql:`!v16` represents a HALF precision vector parameter
+* :sql:`!v32` represents a FLOAT precision vector parameter
+* :sql:`!v64` represents a DOUBLE precision vector parameter
+
+Example insertions (using test DSL syntax for illustration):
+
+.. code-block:: sql
+
+    -- Insert a HALF precision vector
+    INSERT INTO embeddings VALUES (1, !! !v16 [0.5, 1.2, -0.8] !!);
+
+    -- Insert a FLOAT precision vector
+    INSERT INTO embeddings VALUES (2, !! !v32 [0.5, 1.2, -0.8] !!);
+
+    -- Insert a DOUBLE precision vector
+    INSERT INTO embeddings VALUES (3, !! !v64 [0.5, 1.2, -0.8] !!);
+
+Casting Arrays to Vectors
+--------------------------
+
+While vector literals are not supported, you can use :sql:`CAST` to convert array expressions to vectors. The source array elements can be of any numeric type (:sql:`INTEGER`, :sql:`BIGINT`, :sql:`FLOAT`, :sql:`DOUBLE`):
+
+.. code-block:: sql
+
+    -- Cast FLOAT array to FLOAT vector
+    SELECT CAST([1.2, 3.4, 5.6] AS VECTOR(3, FLOAT)) AS vec;
+
+    -- Cast INTEGER array to HALF vector
+    SELECT CAST([1, 2, 3] AS VECTOR(3, HALF)) AS vec;
+
+    -- Cast mixed numeric types to DOUBLE vector
+    SELECT CAST([1, 2.5, 3L] AS VECTOR(3, DOUBLE)) AS vec;
+
+The array must have exactly the same number of elements as the vector's declared dimension, or the cast will fail with error code :sql:`22F3H`. Only numeric arrays can be cast to vectors.
+
+Querying Vectors
+----------------
+
+Vectors can be selected and compared like other column types. When comparing vectors in WHERE clauses, you would typically use prepared statement parameters in your Java/JDBC code:
+
+.. code-block:: sql
+
+    -- Select vectors
+    SELECT embedding FROM embeddings WHERE id = 1;
+
+    -- Compare vectors for equality (using test DSL syntax; in actual code use PreparedStatement)
+    SELECT id FROM embeddings WHERE embedding = !! !v32 [0.5, 1.2, -0.8] !!;
```

Comment:
If we want to restrict DSL syntax to the developer guide, you could write a cast here instead.

Reply:
Or simply write `embedding = ?`